sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -154,6 +155,24 @@ private static String encode(final String stringToEncode) {
         }
     }
 
+    /**
+     * Performs a safe encoding of a url string, only encoding the path.
+     *
+     * @param url The url to encode.
+     * @return The encoded url.
+     */
+    public static String encodeUrlPath(String url) {
+        /* Deconstruct the URL and reconstruct it making sure the path is encoded. */
+        UrlBuilder builder = UrlBuilder.parse(url);
+        String path = builder.getPath();
+        if (path.startsWith("/")) {
```

Comment:
can we have more `/` in the path ? if yes then should we encode them?

Reply:
This particular if case is because when you call getPath, it returns the path including the starting /. Then when you do setPath for some reason, it makes the URL have two slashes - so I'm just trying to prevent that error from happening. Does that make sense?

Reply:
That's not what I meant. Please take a look on this sample url https://github.com/Azure/azure-sdk-for-java/blob/6e23e4acf9b639a87da2bd0f95923c7370290dc8/sdk/storage/azure-storage-blob/src/test/resources/session-records/BlobAPITestcopysourceac%5B4%5D.json#L95 
`https://azstoragesdkaccount.blob.core.windows.net/jtccopysourceac0blobapitestcopysourceac26775340db315343/javablobcopysourceac1blobapitestcopysourceac26713808da3bf` .
the last `/` in this url is going to be encoded into `%2F` I think. Shouldn't we split path by `/` before encoding and just transform segments ?

Reply:
Oh. Both are equivalent as per the service so it shouldnt matter

Reply:
Do we have a test covering that?

Reply:
The fact that the tests work is validation that they are the same since otherwise the service would come back with a 404 saying the source doesnt exist.